openedx/features/course_experience/views/course_home_messages.py

```diff
@@ -55,31 +63,37 @@ def render_to_fragment(self, request, course_id, user_access, **kwargs):
         }
 
         # Register the course home messages to be loaded on the page
-        self.register_course_home_messages(request, course, user_access, course_start_data)
+        self.register_course_home_messages(self, request, course_id, user_access, course_start_data)
 
         # Grab the relevant messages
         course_home_messages = list(CourseHomeMessages.user_messages(request))
 
-        # Return None if user is enrolled and course has begun
-        if user_access['is_enrolled'] and already_started:
-            return None
+        # Pass in the url used to set a course goal
+        set_goal_url = reverse(
+            'openedx.course_experience.set_course_goal', kwargs={
+                'course_id': course_id,
+            }
+        )
 
         # Grab the logo
         image_src = "course_experience/images/home_message_author.png"
 
         context = {
             'course_home_messages': course_home_messages,
+            'set_goal_url': set_goal_url,
             'image_src': image_src,
         }
 
         html = render_to_string('course_experience/course-messages-fragment.html', context)
         return Fragment(html)
 
     @staticmethod
-    def register_course_home_messages(request, course, user_access, course_start_data):
+    def register_course_home_messages(self, request, course_id, user_access, course_start_data):
```

Comment:
We don't want to be adding `self` here.  It looks like you are transitioning this to no longer be static (i.e. drop `@staticmethod` decorator).  I need to think if that makes sense.